utils/python/CIME/compare_test_results.py

```diff
@@ -0,0 +1,86 @@
+import CIME.compare_namelists, CIME.simple_compare
+from CIME.utils import expect
+from CIME.XML.machines import Machines
+from CIME.test_status import *
+from CIME.hist_utils import compare_baseline
+from CIME.case import Case
+
+import os, glob
+
+_MACHINE = Machines()
+
+###############################################################################
+def compare_history(testcase_dir_for_test, baseline_name):
+###############################################################################
+    with Case(testcase_dir_for_test) as case:
+        baseline_full_dir = os.path.join(case.get_value("BASELINE_ROOT"), case.get_value("COMPILER"), baseline_name, case.get_value("CASEBASEID"))
```

Comment:
Should have read your post more closely. Since this path system is built into test_scheduler, then CESM must be using it already. I guess they like it since I haven't seen any complaints ;)